### PR TITLE
Fix BlockStoreQueue test case - a_cut_off_miner...

### PR DESCRIPTION
**Pull request by quantumagi**
See #3041.

This is an alternative fix to PR #3100.

### CODE REVIEW

**Comment by fassadlr** [comment]
@quantumagi with this you can actually just replace the current `IBlockStoreQueueFlushCondition` instance in the node's DI collection that always returns true.

There's an extension method called `ReplaceService` in the integration tests project.

**Comment by quantumagi** [comment]
The existing solution more closely resembles normal operations and is therefore a more plausible test case.

**Comment by fassadlr** [comment]
@quantumagi Still its better for us to not have to override that value every time we want to test functionality like this. Also, changing a `const` to a "settable" value is a design change that needs to be considered more closely.

Also, the reason we implemented that interface was exactly for mocking purposes like this ;)

**Comment by quantumagi** [comment]
@fassadlr, the service override mechanism is broken, and we will need a separate PR to fix that. For now this is the best solution. You can look `ConsensusManager_Reorg_To_Longest_Chain_Multiple_Times_Without_Invalid_Blocks` test case to see why I say it is broken: The `BlockStoreQueueFlushConditionReorgTests`'s `ShouldFlush` is not executing. This seems to be due to the `AddSingleton` in `UseBlockStore` executing after the service override.

**Comment by quantumagi** [comment]
It looks like I would have to fix `ReplaceService` first as it only looks at the `BaseFeature`.

**Comment by fassadlr** [comment]
@quantumagi let me look into why ReplaceService is not working :)

**Comment by quantumagi** [comment]
@fassadlr , see https://github.com/stratisproject/StratisBitcoinFullNode/issues/3169.

**Comment by fassadlr** [comment]
@dangershony my issue here is that we have methods in integration test common projects (extensions) to replace or override services for our integration tests. We can easily replace IBlockStoreFlushCondition here with a version that just returns true everytime. However, @quantumagi has pointed out that the `ReplaceService` doesn't work. I will investigate that now.

**Comment by fassadlr** [comment]
@quantumagi please merge latest from master, I have fixed `ReplaceService`. Can you please try the test again?

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class BlockStoreQueue : IBlockStoreQueue
     {
         /// <summary>Maximum interval between saving batches.</summary>
         /// <remarks>Interval value is a prime number that wasn't used as an interval in any other component. That prevents having CPU consumption spikes.</remarks>
-        private const int BatchMaxSaveIntervalSeconds = 37;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs]: Just revert this then we are good to go!

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs]: Done.

**Pull request merged**